### DISCUSSION TITLE
RL notebook

### DISCUSSION THREAD

**Discussion by fteicht** [erachelson/seq_dec_mak#3] [General]
Discussion about the [RL notebook](https://github.com/erachelson/seq_dec_mak/blob/3716ab2d0dd2d1ad374f6000c9f7129f9e576cfa/rl/rl.ipynb)

**Comment by fteicht** [comment]
I have begun to implement the gym environments for the RL notebook:
- one simple control problem which can be seen as a simplistic version of the aircraft control problem, which helps understand how Gym works
- one aircraft control problem relying on [gym-jsbsim](https://github.com/galleon/gym-jsbsim) for which the Gym environment is hidden in the library

@erachelson Maybe you can begin populate it with the theory of RL and with a simple RL algorithm. I plan to run a RLlib algorithm as well at least to test the environments, but I think we said that we want the students to implement their own RL algorithm in the notebook, right?

**Comment by fteicht** [comment]
@erachelson I saw your RL notebook, thanks!
What's the next step? Should we synchronise your RL notebook with the one I provided with flight control gym envs?
The first `line control` gym env I provided is a simplistic control problem consisting in following a central line, which can be seen as a simplistic introduction to the real `jsbsim`-based flight control env.
Therefore, the `line control` gym env could maybe be used in place of the gym examples you provide in your RL course notebook.
What do tou think @erachelson ?

**Reply by erachelson** [reply]
Let's call each other?
Both options are fine with me.
I find the cartpole example to be easier and more illustrative than the line control problem.
However, in my notebook, I only covered discrete action cases. I intend to give an introduction to different challenges in RL, including continuous actions, in the "theory" part of the class, but didn't think we would have the students implement continuous action control.
Maybe a good option would be to use your notebook as an illustration of what an MDP is, along or after the theory part. This would make for a good transition to a more "hands-on" end of the class, on the line control and the taxiing environments.
